src/controllers/userProfileController.js

```diff
@@ -330,10 +330,12 @@ const userProfileController = function (UserProfile) {
 
   const putUserProfile = async function (req, res) {
     const userid = req.params.userId;
+    const canToggleRequestBio = await hasPermission(req.body.requestor, 'requestBio');
     const isRequestorAuthorized = !!(
       canRequestorUpdateUser(req.body.requestor.requestorId, userid) &&
       ((await hasPermission(req.body.requestor, 'putUserProfile')) ||
-        req.body.requestor.requestorId === userid)
+        req.body.requestor.requestorId === userid ||
+        canToggleRequestBio)
```

Comment:
Hmm. Your suggestion to make a separate endpoint might be the easiest solution to get around this issue. It looks like some fields are protected by `putUserProfileImportantInfo` on line 441, but it will still update the profile's `commonFields` without those changes if you don't have that permission (which could probably mislead people).